### ISSUE TITLE
Ordinal mixed model: no formula-level random cutpoints, and no documented path from a neural predictor into a formula-compiled cumulative link

### ISSUE THREAD

**Issue by aaronstevenwhite** [FACTSlab/quivers#56]
## Context

I am building [LoFI](https://github.com/logical-form-induction/lofi) on quivers. Task 1.1.3 of the
funded NSF CAREER proposal behind it requires **jointly training an ordinal mixed-effects model with
a neural parser**: the model takes a probability of grammaticality (computed by a vector-space chart
parser and type grammar, i.e. a `torch` computation) and produces a distribution over the ordinal
acceptability responses participants gave, with by-participant and by-item effects, trained
end-to-end.

quivers looks like the right home for this: it already has the ordered families, the formula
frontend, SVI/NUTS, and PSIS-LOO for the model comparison the work depends on. This issue records
the three things that stand between that and a working model, since two of them are narrower than
they first appear.

## What already works

`OrderedLogistic` (`src/quivers/continuous/_ordered.py`) is exactly right, and its docstring already
names the target case:

> Per-row cutpoints land in the ordinal-mixed-model setting where every participant has distinct
> thresholds; the distribution's broadcast rules handle both shapes uniformly.

So per-participant thresholds are supported at the distribution level, shared `(K-1,)` and per-row
`(batch, K-1)` cutpoints both broadcast, and the family is registered in `formulas/family.py:198` as
`cumulative -> OrderedLogistic`.

## Gap 1 — `cumulative` does not compile (already tracked as #41)

`family="cumulative"` fails at compile time because `OrderedLogistic` is absent from the DSL
`_FAMILY_REGISTRY`. Noting it here only as a dependency; #41 is the place to fix it.

## Gap 2 — the formula frontend cannot put random effects on the cutpoints

This is the one I could not find a route around. The compiler builds random effects on *coefficients*
(`formulas/compile.py`: non-centered `z_<group>_<slope>`, per-`(group, slope)` scales, indexed plate
draws). I can find no syntax that targets the **thresholds** instead, and grepping
`formulas/*.py` for `cutpoint`, `threshold`, `category.specific` or `cs(` returns nothing.

Per-participant cutpoints are the standard way to model between-annotator scale use, and they are
what the distribution's own docstring anticipates. In brms this is `thres(gr = participant)` plus
category-specific effects; a formula-level equivalent is what is missing. Concretely I want to write
something like:

```python
fit(data, formula="response ~ grammaticality + (1 | item) + (1 | participant)",
    family="cumulative", thresholds_by="participant")
```

and have the compiler emit per-row cutpoints of shape `(batch, K-1)`, which `OrderedLogistic`
already accepts.

If there is an intended route to this through the current surface that I have missed, I would
happily be corrected; that would close this half.

## Gap 3 — composing the ordinal likelihood with an externally computed neural predictor

The linear predictor here is not a design-matrix product. It is the output of a neural chart parser,
and gradients must flow back into it.

`ParamSource` looks like the right seam, and `FunctionSource(fn, param_dim)` ("wraps any `nn.Module`
or callable") and `IdentitySource` ("parameters supplied as data") both look like they would serve.
What I cannot tell from the docs is whether a **formula-compiled** program can have its location term
supplied by a `ParamSource` rather than by formula terms, so that the mixed-effects structure
(random intercepts, random cutpoints) composes with a neural location. Note also that #50 reports
`param_source` reaching only `ConditionalNormal`, though the changelog entries for "Every family that
has a parameter source can select it" read as though that has since been addressed.

The general shape of the request: **a program whose observation model is an ordinal cumulative link
with crossed random effects, whose location comes from an arbitrary `nn.Module`, differentiable
end-to-end.** If that already composes, a worked example would be the most useful possible answer,
and I would contribute one back.

## Prior art to compare against

[`bead`](https://github.com/FACTSlab/bead) implements this model outside quivers, in
`bead/active_learning/models/ordinal_scale.py` with `bead/active_learning/models/random_effects.py`
carrying random intercepts keyed per cutpoint per annotator
(`intercepts["cutpoint_1"]["alice"]`) and variance components per parameter. It is a useful reference
for the structure, and it is the thing I would like to be able to delete in favour of a quivers
program.

## Versions

quivers 0.18.0, Python 3.14.
